contributors/design-proposals/customresources-subresources.md

```diff
@@ -0,0 +1,213 @@
+    // +optional
+    Replicas int32 `json:"replicas,omitempty"`
+}
+
+// ScaleStatus represents the current status of a scale subresource.
+type ScaleStatus struct {
+    // actual number of observed instances of the scaled object.
+    Replicas int32 `json:"replicas"`
+
+    // label query that should match the replicas count. This is same
+    // as the label selector but in the string format to avoid introspection
+    // by clients. The string will be in the same format as the query-param syntax.
+    // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+    // +optional
+    Selector string `json:"selector"`
+}
+```
+
+### Feature Gate
+
+The `SubResources` field in `CustomResourceDefinitionSpec` will be gated under the `CustomResourceSubResources` alpha feature gate.
+If the gate is not open, the value of the new field within `CustomResourceDefinitionSpec` is dropped on creation and updates of CRDs.
+
+## Semantics
+
+### Validation Behavior
+
+#### Status
+
+The status endpoint of a CustomResource receives a full CR object. Changes outside of the `.status` subpath are ignored.
+For validation everything outside of the `.status` subpath can be reset to the existing values of the CustomResource. Then the JSON Schema presented in the CRD is validated against the whole object.
+
+Note: one could extract the status subobject, filter the JSON schema by rules applied to the status and then validate the status subobject only.
+The filter step of the JSON schema though is not obvious in the case that there are status validations not below a properties JSON Schema construct. For that reason, the alternative implementation is preferred as its semantics are much simpler.
+
+#### Scale
+
+Moreover, if the scale subresource is enabled:
+
+- The value at the specified JSON Path `SpecReplicasPath` (e.g. `.spec.replicas`) is validated to ensure that it contains non-negative integer value and is not empty.
+
+- The value at the optional JSON Path `StatusReplicasPath` (e.g. `.status.replicas`) is validated to be a an integer number if it exists (i.e. this can be empty).
+
+- The value at the optional JSON Path `LabelSelectorPath` (e.g. `.spec.labelSelector`) is validated to be a valid label selector if it exists (i.e. this can be empty).
+If it is invalid, an empty `LabelSelector` is used.
+
+### Status Behavior
+
+If the `/status` subresource is enabled, the following behaviors change:
+
+- The main resource endpoint will ignore all changes in the status subpath.
+(note: it will **not** reject requests which try to change the status, following the existing semantics of other resources).
+
+- The `.metadata.generation` field is updated if and only if the value at the `.spec` subpath changes.
+Additionally, if the spec does not change, `.metadata.generation` is not updated.
+
+- The `/status` subresource receives a full resource object, but only considers the value at the `.status` subpath for the update.
```

Comment:
The following is confusing.

> only considers the value at the `.status` subpath for the update.

This already said only `.status` would be considered.

> +The value at the `.metadata` subpath is **not** considered for update

Isn't this unnecessary? Because `.metadata` is obviously not part of `.status`?

Reply:
For legacy reasons, many resources allow updating metadata via the status subresource. That was primarily an accident, but dependent behavior was built around it.

For new resources and CRDs, it is valuable to define status updates more strictly